components/ws-daemon/nsinsider/main.go

```diff
@@ -254,6 +258,193 @@ func main() {
+						return xerrors.Errorf("failed to enable %q: %v", vethIf, err)
+					}
+
+					nc := &nftables.Conn{}
+					nat := nc.AddTable(&nftables.Table{
+						Family: nftables.TableFamilyIPv4,
+						Name:   "nat",
+					})
+
+					postrouting := nc.AddChain(&nftables.Chain{
+						Name:     "postrouting",
+						Hooknum:  nftables.ChainHookPostrouting,
+						Priority: nftables.ChainPriorityNATSource,
+						Table:    nat,
+						Type:     nftables.ChainTypeNAT,
+					})
+
+					// ip saddr 10.0.5.0/24 oifname "eth0" masquerade
+					nc.AddRule(&nftables.Rule{
+						Table: nat,
+						Chain: postrouting,
+						Exprs: []expr.Any{
+							&expr.Payload{
+								DestRegister: 1,
+								Base:         expr.PayloadBaseNetworkHeader,
+								Offset:       12,
+								Len:          net.IPv4len,
+							},
+							&expr.Bitwise{
+								SourceRegister: 1,
+								DestRegister:   1,
+								Len:            net.IPv4len,
+								Mask:           masqueradeAddr.Mask,
+								Xor:            net.IPv4Mask(0, 0, 0, 0),
+							},
+							&expr.Cmp{
+								Op:       expr.CmpOpEq,
+								Register: 1,
+								Data:     masqueradeAddr.IP.To4(),
+							},
+							&expr.Meta{Key: expr.MetaKeyOIFNAME, Register: 1},
+							&expr.Cmp{
+								Op:       expr.CmpOpEq,
+								Register: 1,
+								Data:     []byte(fmt.Sprintf("%s\x00", containerIf)),
+							},
+							&expr.Masq{},
+						},
+					})
+
+					// TODO(toru): we can comment out here when we will remove the slirp4netns and set 10.0.5.1/24 as a default gw
+					// prerouting := nc.AddChain(&nftables.Chain{
+					// 	Name:     "prerouting",
+					// 	Hooknum:  nftables.ChainHookPrerouting,
+					// 	Priority: nftables.ChainPriorityFilter,
+					// 	Table:    nat,
+					// 	Type:     nftables.ChainTypeNAT,
+					// })
+
+					// tcp dport 1-65535 dnat to $cethIp:tcp dport
+					// nc.AddRule(&nftables.Rule{
+					// 	Table: nat,
+					// 	Chain: prerouting,
+					// 	Exprs: []expr.Any{
+					// 		&expr.Meta{Key: expr.MetaKeyL4PROTO, Register: 1},
+					// 		&expr.Cmp{
+					// 			Op:       expr.CmpOpEq,
+					// 			Register: 1,
+					// 			Data:     []byte{unix.IPPROTO_TCP},
+					// 		},
+					// 		&expr.Payload{
+					// 			DestRegister: 1,
+					// 			Base:         expr.PayloadBaseTransportHeader,
+					// 			Offset:       2,
+					// 			Len:          2,
+					// 		},
+
+					// 		&expr.Cmp{
+					// 			Op:       expr.CmpOpGte,
+					// 			Register: 1,
+					// 			Data:     []byte{0x00, 0x01},
+					// 		},
+					// 		&expr.Cmp{
+					// 			Op:       expr.CmpOpLte,
+					// 			Register: 1,
+					// 			Data:     []byte{0xff, 0xff},
+					// 		},
+
+					// 		&expr.Immediate{
+					// 			Register: 2,
+					// 			Data:     cethIp.IP.To4(),
+					// 		},
+					// 		&expr.NAT{
+					// 			Type:        expr.NATTypeDestNAT,
+					// 			Family:      unix.NFPROTO_IPV4,
+					// 			RegAddrMin:  2,
+					// 			RegProtoMin: 1,
+					// 		},
+					// 	},
+					// })
```

Comment:
This is the result of applying everything. I have applied (commented out) everything, changed the default gw, deleted the slirp4netns process, and confirmed that the health check of `supervisor` succeeds and the workspace starts.
```console
root@gitpodio-templategolangc-7xs0qksc7q0:/# nft list table nat
table ip nat {
        chain postrouting {
                type nat hook postrouting priority srcnat; policy accept;
                ip saddr 10.0.5.0/24 oifname "eth0" masquerade
        }

        chain prerouting {
                type nat hook prerouting priority filter; policy accept;
                tcp dport 1-65535 dnat to 10.0.5.2:tcp dport
        }
}
```